File: src/main/java/com/jss/osiris/modules/osiris/miscellaneous/service/TooltipEntryService.java
package com.jss.osiris.modules.osiris.miscellaneous.service;

import java.util.List;

import com.jss.osiris.modules.osiris.miscellaneous.model.TooltipEntry;

public interface TooltipEntryService {
    public List<TooltipEntry> getTooltipEntries();

    public TooltipEntry getTooltipEntry(Integer id);

    public TooltipEntry addOrUpdateTooltipEntry(TooltipEntry tooltipEntry);
}

File: client/src/app/modules/miscellaneous/services/tooltip.entry.service.ts
import { HttpClient, HttpParams } from '@angular/common/http';
import { Injectable } from '@angular/core';
import { AppRestService } from 'src/app/services/appRest.service';
import { TooltipEntry } from '../../miscellaneous/model/TooltipEntry';

@Injectable({
  providedIn: 'root'
})
export class TooltipEntryService extends AppRestService<TooltipEntry> {

  constructor(http: HttpClient) {
    super(http, "miscellaneous");
  }

  getTooltipEntries() {
    return this.getList(new HttpParams(), "tooltip-entries");
  }

  addOrUpdateTooltipEntry(tooltipEntry: TooltipEntry) {
    return this.addOrUpdate(new HttpParams(), "tooltip-entry", tooltipEntry, "Enregistré", "Erreur lors de l'enregistrement");
  }
}